src/coreComponents/linearAlgebra/DofManager.cpp

```diff
@@ -173,9 +174,114 @@ void forMeshSupport( std::vector< DofManager::FieldSupport > const & support,
   }
 }
 
+void fillTrivialPermutation( arrayView1d< localIndex > const permutation )
+{
+  forAll< parallelHostPolicy >( permutation.size(), [&]( localIndex const i )
+  {
+    permutation[i] = i;
+  } );
+}
+
 } // namespace
 
-void DofManager::createIndexArray( FieldDescription const & field )
+array1d< localIndex > DofManager::computePermutation( FieldDescription & field )
+{
+  localIndex const fieldIndex = getFieldIndex( field.name );
+
+  // step 1: save the number of components, and then set it to 1 temporarily
+  //         do not forget to restore at the end
+  //         we set the number of components to 1 to compute the reordering on a smaller matrix
+
+  localIndex const numComps = field.numComponents;
+  CompMask const globallyCoupledComps = field.globallyCoupledComponents;
+  field.numComponents = 1;
+  field.globallyCoupledComponents = CompMask( 1, true );
+
+  // step 2: compute field dimensions (local dofs, global dofs, etc)
+  //         this is needed to make sure that the sparsity pattern function work properly
+  //         in particular, this function defines the rankOffset (computed with number of components = 1)
+
+  computeFieldDimensions( fieldIndex );
+
+  // the number of local dofs is available at this point, we allocate space for the permutation
+  array1d< localIndex > permutation( numLocalDofs( field.name ) );
+
+  // if no reordering is requesting, we just return the identity permutation
+  if( field.reorderingType == LocalReorderingType::None )
+  {
+    fillTrivialPermutation( permutation );
+  }
+  else
+  {
+    fillTrivialPermutation( permutation );
```

Comment:
Is a trivial permutation required as input for the RCM routine? I hadn't studied the algorithm in enough detail, but my previous comments were based on an assumption you wouldn't need to fill it in this branch.

Reply:
Yes, given the way I implemented it, I have to call `createIndexArray` in a first pass to generate the degrees of freedom (assuming `numComps = 1`) here: 
https://github.com/GEOS-DEV/GEOS/blob/e021fa7b3f65b5d129c221db16c9b280eb46a3e0/src/coreComponents/linearAlgebra/DofManager.cpp#L239-L243
I did this to try to reuse the existing code as much as possible, but this is a little bit annoying I agree with you